rebasehelper/tests/test_specfile.py

```diff
@@ -199,6 +200,97 @@ def test_find_archive_target_in_prep(self, spec_object):
         target = spec_object.find_archive_target_in_prep('misc.zip')
         assert target == 'test-1.0.2/misc'
 
+    @pytest.mark.parametrize('spec_attributes, kwargs, expected_content', [
+        (
+            {
+                'save': lambda: None,
+                'removed_patches': [],
+                'spec_content': dedent("""
+                    Patch0:    0.patch
+                    Patch1:    1.patch
+
+                    %patchlist
+                    2.patch
+                    3.patch
+
+                    %prep
+                    %patch0 -p0
+                    %patch1 -p1
+                    %patch2 -p0
+                    %patch3 -p1
+                    """),
+            },
+            {
+                'patches':
+                    {
+                        'deleted': ['2.patch'],
+                        'inapplicable': ['1.patch'],
+                    },
+                'disable_inapplicable': False,
+            },
+            dedent("""
+                Patch0:    0.patch
+                Patch1:    1.patch
+
+                %patchlist
+                3.patch
+
+                %prep
+                %patch0 -p0
+                # The following patch contains conflicts
+                %patch1 -p1
+                %patch2 -p1
+                """)
+        ),
+        (
+            {
+                'save': lambda: None,
+                'removed_patches': [],
+                'spec_content': dedent("""
+                    Patch0:    0.patch
+                    Patch1:    1.patch
+
+                    %patchlist
+                    2.patch
+                    3.patch
+
+                    %prep
+                    %patch0 -p0
+                    %patch1 -p1
+                    %patch2 -p0
+                    %patch3 -p1
+                    """),
+            },
+            {
+                'patches':
+                    {
+                        'deleted': ['2.patch'],
+                        'inapplicable': ['1.patch'],
+                    },
+                'disable_inapplicable': True,
+            },
+            dedent("""
+        Patch0:    0.patch
+        #Patch1:    1.patch
+
+        %patchlist
+        3.patch
+
+        %prep
+        %patch0 -p0
+        # The following patch contains conflicts
+        #%%patch1 -p1
+        %patch1 -p1
+        """),
+        ),
+    ], ids=[
+        'do_not_disable_inapplicable',
+        'disable_inapplicable',
+    ])
+    def test_write_updated_patches(self, kwargs, expected_content, mocked_spec_object):
```

Comment:
Just a nitpick, but I would prefer `mocked_spec_object` to be the first argument, since it is being used/accessed first.

Reply:
Also, it would match the order of parametrized arguments (`spec_attributes` directly influences `mocked_spec_object`).